twitter-clone/src/AuthContext.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { createContext, useState, useContext } from 'react';
+
+export interface User {
+  username: string;
+  id: string;
+  name: string;
+  email: string;
+}
+
+interface AuthContextType {
+  user: User | null;
+  logIn: (userData: User) => void;
+  logOut: () => void;
+}
```

Comment:
Good idea to store types in separayte `types.ts` file.

If you don't need to use interface features like `implements` or other is beeter to use type instead.
Especially in context of this type: AuthContextType it's already named as Type, little confusing.

Reply:
Moved shared types.